src/test/java/edu/ucla/library/bucketeer/utils/FilesystemWriteCsvFfOnT.java

```diff
@@ -0,0 +1,180 @@
+                        aContext.assertTrue(tmpDestDir.exists() || tmpDestDir.mkdirs());
+
+                        // Confirm we can copy the test container's files to the temporary test directory
+                        aContext.assertTrue(
+                                DockerUtils.copy(TestConstants.BUCKETEER_FF_ON, srcDir, tmpDestDir.toString()));
+
+                        // Confirm the file we expect to exist actually does
+                        aContext.assertTrue(expectedFile.exists());
+
+                        FileUtils.delete(expectedFile);
+                        asyncTask.complete();
+                    } else {
+                        aContext.fail(fakeLambda.cause());
+                    }
+                });
+                fakeLambda(webClient, port, Constants.UNSPECIFIED_HOST, jobCompletion);
+            } else {
+                aContext.fail(sendForm.cause());
+            }
+        });
+    }
+
+    /**
+     * Mock the response from AWS Lambda in order to complete the job.
+     *
+     * @param aWebClient A client used to send requests to our Bucketeer instance
+     * @param aPort The port of our Bucketeer instance
+     * @param aHost The host of our Bucketeer instance
+     * @param aPromise A promise to complete upon job completion, or fail
+     */
+    private void fakeLambda(final WebClient aWebClient, final int aPort, final String aHost,
+            final Promise<Void> aPromise) {
+        Future.<Void>future(s3BucketVerticleSwap -> {
+            s3BucketVerticleSwap.complete();
+        }).compose(success -> {
+            return Future.<List<String>>future(getImageIds -> {
+                aWebClient.get(aPort, aHost, "/batch/jobs/live-test-docker").send(ar -> {
+                    if (ar.succeeded()) {
+                        // Gets a list of the image IDs of the current jobs
+                        final List<String> imageIds = ar.result().bodyAsJsonObject().getJsonArray("jobs").stream()
+                                .map(JsonObject.class::cast).filter(job -> job.getString("status").equals(""))
+                                .map(job -> job.getString("image-id")).collect(Collectors.toList());
+
+                        getImageIds.complete(imageIds);
+                    } else {
+                        getImageIds.fail(ar.cause());
+                    }
+                });
+            });
+        }).compose(imageIds -> {
+            final List<Future> futures = imageIds.stream().map(imageId -> {
+                return Future.<Void>future(finishJob -> {
+                    final String urlEncodedImageId = URLEncoder.encode(imageId, StandardCharsets.UTF_8);
+                    final String urlPath = "/batch/jobs/live-test-docker/" + urlEncodedImageId + "/true";
+
+                    // Tell our Bucketeer instance to treat the job as completed
+                    aWebClient.patch(aPort, aHost, urlPath).send(ar -> {
```

Comment:
I know Vertx uses `ar` in all their examples, but I find that bit cryptic, personally... could be `asyncResult`? I personally also like the action verb (`patch` in this case... which gives you `patch.succeeded()` and `patch.cause()` (which a little less clear, I know)). Not a requirement... just expressing an opinion on naming.

Reply:
I agree, `ar` isn't good. I can change